adsdata/reader.py

```diff
@@ -102,8 +121,9 @@ def read(self, size=-1):
     
 
     def readline(self):
-        line = self._iostream.readline()
-        return self.process_line(line)
+        return self.read()
```

Comment:
read() returns contents of the whole line (or requested number of bytes) - readline returns just one line

so you may want to change this PR